web-crawler/src/main/java/com/google/sps/webcrawler/WebCrawler.java

```diff
@@ -132,65 +144,74 @@ public String handleResponse(final HttpResponse response) throws IOException {
       Gson gson = new Gson();
       Object jsonResponse = gson.fromJson(responseBody, Object.class);
       //@TODO [Unpack {@code jsonResponse} and find URLs.]
-      return Arrays.asList(
-          "https://www.cnn.com/2020/06/23/politics/aoc-ny-primary-14th-district/index.html");
+      return Arrays.asList(TEST_URL);
     } catch (IOException e){
       System.out.println("[ERROR] Error occurred with fetching URLs from Custom Search: " + e);
-      return Arrays.asList(
-          "https://www.cnn.com/2020/06/23/politics/aoc-ny-primary-14th-district/index.html");
+      return Arrays.asList(TEST_URL);
     }
   }
 
   /** 
    * Checks robots.txt for permission to web-scrape, scrapes webpage if permitted and extracts
-   * textual content.
+   * textual content. Returns an empty {@code Optional<NewsArticle>} in the event of an exception.
    */
-  public NewsArticle scrapeAndExtractHtml(String url) {
-    String robotsUrl = url.substring(0, url.indexOf("/", url.indexOf("//") + 2) + 1)
-        + "robots.txt";
+  public Optional<NewsArticle> scrapeAndExtractHtml(String url) {
     try {
+      URL formattedUrl = new URL(url);
+      String robotsUrl = String.format("%s://%s/robots.txt", formattedUrl.getProtocol(),
```

Comment:
You could instead use the URL constructor `URL(String protocol, String host, String file)` to get the robots URL object directly.